### PR TITLE
github/workflows: add lua linting

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: player/lua/osc.lua [hunk 1]
```diff
@@ -742,7 +745,7 @@ function render_elements(master_ass)
                                 if pos > range["end"] then
                                     pend = get_slider_ele_pos_for(element, range["end"])
                                 end
-                                elem_ass:rect_ccw(pstart, elem_geo.h - foV - seekRangeLineHeight, pend, elem_geo.h - foV)
+                                elem_ass:rect_ccw(pstart,elem_geo.h - foV - seekRangeLineHeight, pend, elem_geo.h - foV)
```
[the remark this answers is not in the document]
Comment (kasper93) [reply player/lua/osc.lua]: fixed

File: player/lua/defaults.lua [hunk 2]
```diff
@@ -1,5 +1,5 @@
 -- Compatibility shim for lua 5.2/5.3
-unpack = unpack or table.unpack
+local unpack = unpack or table.unpack -- luacheck: globals unpack table.unpack
```
Comment (guidocella) [inline player/lua/defaults.lua:2]: This makes it inaccessible from other scripts.

Comment (kasper93) [reply player/lua/defaults.lua:2]: It is not used in any other scripts, so it is made local. It is not something we should export anyway.

Comment (guidocella) [reply player/lua/defaults.lua:2]: I don't know if 3rd-party scripts may depend on it. But according to https://github.com/mpv-player/mpv/issues/1648 it is only needed if you compile 5.2 without `LUA_COMPAT_UNPACK` anyway. I did use this in the select.lua PR so I'll update that.

Comment (kasper93) [reply player/lua/defaults.lua:2]: We can export it as mp.unpack.

> I don't know if 3rd-party scripts may depend on it.

Well, not our problem. Though indeed, we would silently fix global unpack if it were to be not available.

Comment (guidocella) [reply player/lua/defaults.lua:2]: It's probably fine to just make it local with how rarely it is needed.

**Comment by kasper93** [comment]
I'm done bikesheding this, there are remaining issues in scripts regarding global variables, but they can be resolved individually in next PRs, there is `todo` list of scripts to fix.

**Review by guidocella — APPROVED** [review]

File: .luacheckrc [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            raw_wait_event = {},
+            request_event = {},
+            script_name = {},
+            use_suspend = {},
+            wait_event = {},
+        }
+    }
+}
+
+std = "min+mp"
```
Comment (sfan5) [inline .luacheckrc:102]: not `lua51+mp`?

File: player/lua/defaults.lua (hunk 2 above)
Comment (sfan5) [inline player/lua/defaults.lua:2]: personally I would keep this.

File: player/lua/osc.lua [hunk 4]
```diff
@@ -2205,8 +2208,8 @@ end
 
 function reset_margins()
     if state.using_video_margins then
-        for _, opt in ipairs(margins_opts) do
```
Comment (sfan5) [inline player/lua/osc.lua:2208]: this is one of the shadow/redefinition warnings, right?
I'd just exclude those.

File: .luacheckrc (hunk 3 above)
Comment (kasper93) [reply .luacheckrc:102]: I wanted intersection between `lua51` and `luajit`, but didn't find a way do define like that and `min` which is minimal intersection between all supported version works fine for us too.

File: player/lua/osc.lua (hunk 4 above)
Comment (kasper93) [reply player/lua/osc.lua:2208]: yes, it shadows `opt`. I think it is very easy to miss, I'd like to keep it.

File: player/lua/defaults.lua (hunk 2 above)
Comment (kasper93) [reply player/lua/defaults.lua]: Restored, but for the record, not a big fun of it.

**Comment by kasper93** [comment]
Merging, so we can iterate on this in smaller prs.

**Pull request merged**